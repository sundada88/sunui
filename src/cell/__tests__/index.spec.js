import Cell from '../index'
import example from '../example'
import SunCell from '../cell'
import { createApp } from 'vue'
import { mount } from '@vue/test-utils'

test('test cell example', () => {
  const wrapper = mount(example)
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.unmount()
})

test('test cell plugin', () => {
  const app = createApp({}).use(Cell)
  expect(app.component(Cell.name)).toBeTruthy()
})

test('test cell props', () => {
  const title = 'This is Cell'
  const desc = 'This is desc'
  const iconClass = 'test-icon-class'
  const titleClass = 'test-title-class'
  const descClass = 'test-desc-class'
  const extraClass = 'test-extra-class'
  const icon = 'fire'
  const wrapper = mount(SunCell, {
    props: {
      title,
      desc,
      border: true,
      iconClass,
      titleClass,
      descClass,
      extraClass,
      icon
    }
  })
  expect(wrapper.classes('sun-cell')).toBeTruthy()
  expect(wrapper.find('.sun-icon').classes('sun-icon-fire')).toBeTruthy()
  expect(wrapper.classes('sun-cell--border')).toBeTruthy()
  expect(wrapper.find('.sun-cell__title').text()).toBe(title)
  expect(wrapper.find('.sun-cell__desc').text()).toBe(desc)
  expect(wrapper.find(iconClass)).toBeTruthy()
  expect(wrapper.find(titleClass)).toBeTruthy()
  expect(wrapper.find(descClass)).toBeTruthy()
  expect(wrapper.find(extraClass)).toBeTruthy()
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.unmount()
})
