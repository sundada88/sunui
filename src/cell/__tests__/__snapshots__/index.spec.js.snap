// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test cell example 1`] = `
"<div>
  <div class=\\"app-type\\">基本使用</div>
  <div class=\\"sun-cell\\">
    <div class=\\"sun-cell__content\\">
      <div class=\\"sun-cell__title\\">这是单元格</div>
      <div class=\\"sun-cell__desc\\">
        <!---->
      </div>
    </div>
    <div class=\\"sun-cell__extra\\"> </div>
  </div>
  <div class=\\"sun-cell\\">
    <div class=\\"sun-cell__content\\">
      <div class=\\"sun-cell__title\\">这是单元格</div>
      <div class=\\"sun-cell__desc\\">
        <!---->
      </div>
    </div>
    <div class=\\"sun-cell__extra\\"> </div>
  </div>
  <div class=\\"app-type\\">显示图标</div>
  <div class=\\"sun-cell\\">
    <div class=\\"sun-cell__icon\\"><i class=\\"sun-icon sun-icon--set sun-icon-fire sun-flex\\" style=\\"transition: all 0ms;\\"></i></div>
    <div class=\\"sun-cell__content\\">
      <div class=\\"sun-cell__title\\">这是单元格</div>
      <div class=\\"sun-cell__desc\\">
        <!---->
      </div>
    </div>
    <div class=\\"sun-cell__extra\\"> <i class=\\"sun-icon sun-icon--set sun-icon-information\\" style=\\"transition: all 0ms;\\"></i></div>
  </div>
  <div class=\\"sun-cell\\">
    <div class=\\"sun-cell__icon\\"><i class=\\"sun-icon sun-icon--set sun-icon-fire\\" style=\\"transition: all 0ms;\\"></i></div>
    <div class=\\"sun-cell__content\\">
      <div class=\\"sun-cell__title\\">这是单元格</div>
      <div class=\\"sun-cell__desc\\">
        <!---->
      </div>
    </div>
    <div class=\\"sun-cell__extra\\"> <i class=\\"sun-icon sun-icon--set sun-icon-information\\" style=\\"transition: all 0ms;\\"></i></div>
  </div>
  <div class=\\"app-type\\">显示描述</div>
  <div class=\\"sun-cell\\">
    <div class=\\"sun-cell__icon\\"><i class=\\"sun-icon sun-icon--set sun-icon-fire sun-flex\\" style=\\"transition: all 0ms;\\"></i></div>
    <div class=\\"sun-cell__content\\">
      <div class=\\"sun-cell__title\\">这是单元格</div>
      <div class=\\"sun-cell__desc\\">描述</div>
    </div>
    <div class=\\"sun-cell__extra\\"> <i class=\\"sun-icon sun-icon--set sun-icon-information\\" style=\\"transition: all 0ms;\\"></i></div>
  </div>
  <div class=\\"sun-cell\\">
    <div class=\\"sun-cell__icon\\"><i class=\\"sun-icon sun-icon--set sun-icon-fire sun-flex\\" style=\\"transition: all 0ms;\\"></i></div>
    <div class=\\"sun-cell__content\\">
      <div class=\\"sun-cell__title\\">这是单元格</div>
      <div class=\\"sun-cell__desc\\">描述</div>
    </div>
    <div class=\\"sun-cell__extra\\"> <i class=\\"sun-icon sun-icon--set sun-icon-information\\" style=\\"transition: all 0ms;\\"></i></div>
  </div>
  <div class=\\"sun-cell\\">
    <div class=\\"sun-cell__content\\">
      <div class=\\"sun-cell__title\\">这是单元格</div>
      <div class=\\"sun-cell__desc\\">描述</div>
    </div>
    <div class=\\"sun-cell__extra\\"> <i class=\\"sun-icon sun-icon--set sun-icon-information\\" style=\\"transition: all 0ms;\\"></i></div>
  </div>
  <div class=\\"app-type\\">显示边框</div>
  <div class=\\"sun-cell sun-cell--border\\">
    <div class=\\"sun-cell__content\\">
      <div class=\\"sun-cell__title\\">这是单元格</div>
      <div class=\\"sun-cell__desc\\">
        <!---->
      </div>
    </div>
    <div class=\\"sun-cell__extra\\"> </div>
  </div>
  <div class=\\"sun-cell sun-cell--border\\">
    <div class=\\"sun-cell__content\\">
      <div class=\\"sun-cell__title\\">这是单元格</div>
      <div class=\\"sun-cell__desc\\">
        <!---->
      </div>
    </div>
    <div class=\\"sun-cell__extra\\"> </div>
  </div>
</div>"
`;

exports[`test cell props 1`] = `
"<div class=\\"sun-cell sun-cell--border\\">
  <div class=\\"sun-cell__icon iconClass\\"><i class=\\"sun-icon sun-icon--set sun-icon-fire sun-flex\\" style=\\"transition: all 0ms;\\"></i></div>
  <div class=\\"sun-cell__content\\">
    <div class=\\"sun-cell__title test-title-class\\">This is Cell</div>
    <div class=\\"sun-cell__desc test-desc-class\\">This is desc</div>
  </div>
  <div class=\\"sun-cell__extra test-extra-class\\"> </div>
</div>"
`;
